src/plugins.js

```diff
@@ -30,62 +30,62 @@ export function setCsrfCookies(request, reply) {
  * function calls `server.state` for both cookie names before registering the
  * plugin.
  *
- * @param {String} secret the 'salt' for encoding the CSRF tokens
- * @return {Object} the { register, options } object for a `server.register` call.
+ * @param {Object} config the environment configuration object
+ * @return {Object} the { register } object for a `server.register` call.
  */
-export function getCsrfPlugin(secret) {
+export function getCsrfPlugin(config) {
 	const register = (server, options, next) => {
 		const cookieOptions = {
 			path: '/',
-			isSecure: process.env.NODE_ENV === 'production',
+			isSecure: options.isProd,
 		};
+
 		server.state(
-			'x-csrf-jwt',  // set by plugin
-			{ ...cookieOptions, isHttpOnly: true }  // no client-side interaction needed
+			'x-csrf-jwt', // set by plugin
+			{ ...cookieOptions, isHttpOnly: true } // no client-side interaction needed
 		);
+
 		server.state(
-			'x-csrf-jwt-header',  // set by onPreResponse
+			'x-csrf-jwt-header', // set by onPreResponse
 			{ ...cookieOptions, isHttpOnly: false } // the client must read this cookie and return as a custom header
 		);
 
 		const registration = CsrfPlugin.register(server, options, next);
-		server.ext('onPreResponse', setCsrfCookies);  // this extension must be registered _after_ plugin is registered
+		server.ext('onPreResponse', setCsrfCookies); // this extension must be registered _after_ plugin is registered
 
 		return registration;
 	};
+
 	register.attributes = CsrfPlugin.register.attributes;
+
 	return {
 		register,
 		options: {
-			secret,
-		}
+			secret: config.csrf_secret,
+			isProd: config.isProd,
```

Comment:
Same method from last comment -  will fix, but would like to understand why `server.app` didn't work, but `server.settings.app` does now